### DISCUSSION TITLE
The VirtualBox Linux kernel driver is not loaded or not set up correctly

### DISCUSSION THREAD

**Discussion by kabeza** [elementary/triage#291] [Q&A] [answered]
Hi
I've searched and found thread #272 but I think mine could be another problem because I don't have any NVidia driver installed.
I've installed the Virtualbox 7.0.10 (downloading deb file) for Jammy 22.04 (Horus equivalent)
I can run VirtualBox, etc. but when I want to run a VM I get the following

![imagen_1528](https://github.com/elementary/triage/assets/274856/7b668dfd-ba14-4951-9469-69f96dddbf72)

![imagen_1529](https://github.com/elementary/triage/assets/274856/0326f48a-90d8-4e9c-954d-c373d62e8d84)

Then, I try to execute the suggested command:
`sudo /sbin/vboxconfig`  

And I get:

> vboxdrv.sh: Stopping VirtualBox services.
> vboxdrv.sh: Starting VirtualBox services.
> vboxdrv.sh: Building VirtualBox kernel modules.
> This system is currently not set up to build kernel modules.
> Please install the Linux kernel "header" files matching the current kernel
> for adding new hardware support to the system.
> The distribution packages containing the headers are probably:
>     linux-headers-generic linux-headers-6.2.0-26-generic
> This system is currently not set up to build kernel modules.
> Please install the Linux kernel "header" files matching the current kernel
> for adding new hardware support to the system.
> The distribution packages containing the headers are probably:
>     linux-headers-generic linux-headers-6.2.0-26-generic
> 
> There were problems setting up VirtualBox.  To re-start the set-up process, run
>   /sbin/vboxconfig
> as root.  If your system is using EFI Secure Boot you may need to sign the
> kernel modules (vboxdrv, vboxnetflt, vboxnetadp, vboxpci) before you can load
> them. Please see your Linux system's documentation for more information.
> 

After this, I've tried removing completely virtualbox-7.0 package, reboot, and then to try the manual installation procedure:  
- download and add the oracle public key to the keyring
- add the corresponding line to sources.list
- `deb [arch=amd64 signed-by=/usr/share/keyrings/oracle-virtualbox-2016.gpg] https://download.virtualbox.org/virtualbox/debian jammy contrib`
- sudo apt update, etc.
- sudo apt install virtualbox-6.1

I execute Virtualbox, etc. but when I try to run a VM I get the same problem...

I've read some articles where suggest changing grub config, install new kernels, etc. but I don't want to screw up this fresh Elementary OS install

`uname -a`
> Linux kabezapc 6.2.0-26-generic #26~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Thu Jul 13 16:27:29 UTC 2 x86_64 x86_64 x86_64 GNU/Linux

`dkms status`

> virtualbox/6.1.38: added

**PS: I've EFI Secure Boot disabled**

Thanks for any suggestion

**Comment by lenemter** [comment]
> but I think mine could be another problem because I don't have any NVidia driver installed

That doesn't matter, #272 is about installing linux development headers, which according to your logs, virtualbox is asking for. So just try to follow steps in #272 and see what happens.

**Comment by davidmhewitt** [comment, marked as answer]
The solution is probably what's suggested in the error message:

Install the kernel headers.

`sudo apt install linux-headers-generic-hwe-22.04`

After this, you may need to reboot and/or reinstall Virtualbox.

**Reply by kabeza** [reply]
This shouldn't screw up with grub or anything that may break the Elementary OS install, right ?

Thanks

**Reply by davidmhewitt** [reply]
Nope, it's safe to install that package on any elementary or Ubuntu install. That package doesn't actually do anything by itself. It just provides support files to allow other packages to install drivers into your running Linux kernel.

But after you do it, Virtualbox will be able to install drivers into your install, so if anything is going to break your install, it'll be VirtualBox 😆 

Should be fine though!

**Comment by kabeza** [comment]
- Removed everything related to Virtualbox
- Installed (as suggested by both)
- `sudo apt install linux-headers-generic-hwe-22.04`
- Reboot, reinstall VirtualBox 7 deb

All working fine. Thanks
